pandas/core/internals.py

```diff
@@ -2930,621 +2513,438 @@ def consolidate(self):
+        """
+        Slice/take blocks along axis=0.
 
-        return self.__class__(new_blocks, new_axes)
+        Overloaded for SingleBlock
 
-    def reindex_items(self, new_items, indexer=None, copy=True,
-                      fill_value=None):
-        """
+        Returns
+        -------
+        new_blocks : list of Block
 
         """
-        new_items = _ensure_index(new_items)
-        data = self
-        if not data.is_consolidated():
-            data = data.consolidate()
-            return data.reindex_items(new_items, copy=copy,
-                                      fill_value=fill_value)
 
-        if indexer is None:
-            new_items, indexer = self.items.reindex(new_items,
-                                                    copy_if_needed=True)
-        new_axes = [new_items] + self.axes[1:]
+        allow_fill = fill_tuple is not None
 
-        # could have so me pathological (MultiIndex) issues here
-        new_blocks = []
-        if indexer is None:
-            for blk in self.blocks:
-                if copy:
-                    blk = blk.reindex_items_from(new_items)
-                else:
-                    blk.ref_items = new_items
-                new_blocks.extend(_valid_blocks(blk))
-        else:
+        sl_type, slobj, sllen = _preprocess_slice_or_indexer(
+            slice_or_indexer, self.shape[0], allow_fill=allow_fill)
 
-            # unique
-            if self.axes[0].is_unique and new_items.is_unique:
+        if self._is_single_block:
+            blk = self.blocks[0]
 
-                # ok to use the global indexer if only 1 block
-                i = indexer if len(self.blocks) == 1 else None
+            if sl_type in ('slice', 'mask'):
+                return [blk.getitem_block(slobj,
+                                          new_mgr_locs=slice(0, sllen))]
+            elif not allow_fill or self.ndim == 1:
+                if allow_fill and fill_tuple[0] is None:
+                    _, fill_value = com._maybe_promote(blk.dtype)
+                    fill_tuple = (fill_value,)
+
+                return [blk.take_nd(slobj, axis=0,
+                                    new_mgr_locs=slice(0, sllen),
+                                    fill_tuple=fill_tuple)]
+
+        if sl_type in ('slice', 'mask'):
+            blknos = self._blknos[slobj]
+            blklocs = self._blklocs[slobj]
+        else:
+            blknos = com.take_1d(self._blknos, slobj, fill_value=-1,
+                                 allow_fill=allow_fill)
+            blklocs = com.take_1d(self._blklocs, slobj, fill_value=-1,
+                                  allow_fill=allow_fill)
+
+        # When filling blknos, make sure blknos is updated before appending to
+        # blocks list, that way new blkno is exactly len(blocks).
+        #
+        # FIXME: mgr_groupby_blknos must return mgr_locs in ascending order,
+        # pytables serialization will break otherwise.
```

Comment:
@jreback long-shot any idea if this FIXME is still needed or what it would take to address?

Reply:
no idea
sorry

Reply:
This PR is sure quite a trip down the memory lane :) 

Unfortunately, I don't quite remember what was the problem with pytables serialisation here, but looks like mgr_locs are returned in ascending order from `lib.get_blkno_indexers` anyway, so it's likely that the FIXME is not needed anymore.